### ISSUE TITLE
Unable to successfully clear the partitions after OS is installed

### ISSUE THREAD

**Issue by whowutwut** [xcat2/xcat-core#1080]
I'm not sure if this is only after Ubuntu is installed, but we would probably need to focus to make sure that we can successfully clear the partitions when we are switching OS.   I thought there was some Issue that is similar to this but I can't locate any.... 

In the Power S822LC machines, I believe the following events happened to get us into this situation.
1 Ubuntu 16.04 is installed on the machine
2 Attempt to switch the OS to PowerKVM 3.1, it looks like the PowerKVM OS installs OK but when the OS loads, it's back into the Ubuntu OS.  
```
IBM_PowerKVM release 3.1.0 gold build 41.0 (pkvm3_1_0)
Kernel 3.18.22-359.el7_1.pkvm3_1_0.4000.1.ppc64le on a ppc64le (hvc0)
c910f05c39 login: root (automatic login)
         Starting LSB: Start NTP daemon...
         Starting LSB: SNMP agents...
[  OK  ] Started /etc/rc.local Compatibility.
         Starting Terminate Plymouth Boot Screen...
         Starting Hold until boot process finishes up...
[  OK  ] Started Hold until boot process finishes up.
[  OK  ] Started Serial Getty on hvc0.
[  OK  ] Started Getty on tty1.
[  OK  ] Reached target Login Prompts.
[  OK  ] Started Terminate Plymouth Boot Screen.
[  OK  ] Started LSB: Start NTP daemon.
         Mounting Arbitrary Executable File Formats File System...
[  OK  ] Started OpenBSD Secure Shell server.
[  OK  ] Mounted Arbitrary Executable File Formats File System.
[  OK  ] Started LSB: SNMP agents.
[  OK  ] Reached target Multi-User System.
[  OK  ] Reached target Graphical Interface.
[  OK  ] Started Stop ureadahead data collection 45s after completed startup.
         Starting Update UTMP about System Runlevel Changes...
[  OK  ] Started Update UTMP about System Runlevel Changes.
Ubuntu 16.04 LTS c910f05c39 hvc0
c910f05c39 login:
```

Suspect that maybe the disk that PowerKVM was installed on was not the primary disk....  One wild guess is that using the development build with some bug in the `getinstdisk` script, it selected the incorrect disk to install the OS onto and we installed PowerKVM onto the wrong disk.  

3 Tried to install RHEL instead of PowerKVM to see if we could switch the OS and it failed at this point: 

```
================================================================================
Installation

 1) [x] Language settings                 2) [x] Timezone settings
        (English (United States))                (America/New_York timezone)
 3) [x] Installation source               4) [!] Software selection
        (http://10.2.5.3:80/install/rhe          (Minimal Install)
        ls7.2/ppc64le)                    6) [x] Kdump
 5) [!] Installation Destination                 (Kdump is enabled)
        (Error checking storage configu   8) [ ] User creation
        ration)                                  (No user will be created)
 7) [x] Network configuration
        (Wired (enP3p3s0f3) connected)
  Please make your choice from above ['q' to quit | 'b' to begin installation |
  'r' to refresh]:
```

4 Similar to #1037 , updated xCAT to a later build to get the fixes for `getinstdisk`, and then set `xcatdebugmode=2` so we could check the install_disk calculations... 

```
# lsxcatd -v
Version 2.12 (git commit bb5988332333745cbe759f5c8299c8379170b994, built Mon May  9 09:30:39 EDT 2016)
```

Some very strange things observed in anaconda shell: 

```
[anaconda root@c910f05c39 ~]# cat /tmp/install_disk 
/dev/sdk
[anaconda root@c910f05c39 ~]# ls /dev/sd*
/dev/sda  /dev/sdb  /dev/sdc  /dev/sdd	/dev/sde  /dev/sdf  /dev/sdg  /dev/sdh	/dev/sdi  /dev/sdj  /dev/sdj1  /dev/sdj2  /dev/sdj3  /dev/sdk  /dev/sdk1  /dev/sdk2  /dev/sdk3
```

There are only 2 physical disks possible in IBM Power S822LC machines.   So not sure how we got this configuration with 11 disks... 

Then looking at parted, there ARE actually 2 disks... 

```
[anaconda root@c910f05c39 ~]# parted
GNU Parted 3.1
Using /dev/mapper/live-base
Welcome to GNU Parted! Type 'help' to view a list of commands.
(parted) print all                                                        
Model: Linux device-mapper (linear) (dm)
Disk /dev/mapper/live-base: 2147MB
Sector size (logical/physical): 512B/512B
Partition Table: loop
Disk Flags: 

Number  Start  End     Size    File system  Flags
 1      0.00B  2147MB  2147MB  ext4


Model: Linux device-mapper (snapshot) (dm)
Disk /dev/mapper/live-rw: 2147MB
Sector size (logical/physical): 512B/512B
Partition Table: loop
Disk Flags: 

Number  Start  End     Size    File system  Flags
 1      0.00B  2147MB  2147MB  ext4


Model: ATA ST1000NX0313 (scsi)
Disk /dev/sdj: 1000GB
Sector size (logical/physical): 512B/4096B
Partition Table: gpt
Disk Flags: 

Number  Start   End     Size    File system     Name  Flags
 1      1049kB  8389kB  7340kB                        prep
 2      8389kB  998GB   998GB   ext4
 3      998GB   1000GB  2048MB  linux-swap(v1)


Model: ATA ST1000NX0313 (scsi)
Disk /dev/sdk: 1000GB
Sector size (logical/physical): 512B/4096B
Partition Table: gpt
Disk Flags: pmbr_boot

Number  Start   End     Size    File system  Name  Flags
 1      1049kB  9437kB  8389kB                     prep
 2      9437kB  546MB   537MB   ext4
 3      546MB   1000GB  1000GB                     lvm
```

Looks like some of the `/dev/sdk` is already partitioned to prepare for PowerKVM ... 

So I think the conclusion here is that the installer has some trouble clearing out existing partitions.  I'd like to make some changes to the kickstart template file to see if we can recover from this situation, rather than using `parted` to force a clean re-partition and then starting again. 

@xuweibj @tingtli Thoughts?

**Comment by whowutwut** [comment]
Some more information, when I'm looking at the `zerombr` and `clearpart --all` documentation for RHEL, I am seeing that clearpart is executed first in the kickstart and then zerombr..  . I made this test change and re-ran the provisioning..  it seemed to be slightly better but still not selecting `/dev/sda`.... 

```
[anaconda root@c910f05c39 tmp]# cat install_disk 
/dev/sdb
[anaconda root@c910f05c39 tmp]# parted
GNU Parted 3.1
Using /dev/sda
Welcome to GNU Parted! Type 'help' to view a list of commands.
(parted) print all                                                        
Model: ATA ST1000NX0313 (scsi)
Disk /dev/sda: 1000GB
Sector size (logical/physical): 512B/4096B
Partition Table: gpt
Disk Flags: 

Number  Start   End     Size    File system     Name  Flags
 1      1049kB  8389kB  7340kB                        prep
 2      8389kB  998GB   998GB   ext4
 3      998GB   1000GB  2048MB  linux-swap(v1)


Model: ATA ST1000NX0313 (scsi)
Disk /dev/sdb: 1000GB
Sector size (logical/physical): 512B/4096B
Partition Table: gpt
Disk Flags: pmbr_boot

Number  Start   End     Size    File system  Name  Flags
 1      1049kB  9437kB  8389kB                     prep
 2      9437kB  546MB   537MB   ext4
 3      546MB   1000GB  1000GB                     lvm


Model: Linux device-mapper (linear) (dm)
Disk /dev/mapper/live-base: 2147MB
Sector size (logical/physical): 512B/512B
Partition Table: loop
Disk Flags: 

Number  Start  End     Size    File system  Flags
 1      0.00B  2147MB  2147MB  ext4


Model: Linux device-mapper (snapshot) (dm)
Disk /dev/mapper/live-rw: 2147MB
Sector size (logical/physical): 512B/512B
Partition Table: loop
Disk Flags: 

Number  Start  End     Size    File system  Flags
 1      0.00B  2147MB  2147MB  ext4
```

The code I'm running does have @zet809 's pull request #965 which I hit before...

**Comment by whowutwut** [comment]
Even more interesting things in my debug....  I created a script that would brute force remove the partitions on the physical disk... 

```
[root@c910f02c05p03 share]# diff ./xcat/install/scripts/pre.rh.rhels7 ./xcat/install/scripts/pre.rh.rhels7.orig
6,13d5
< #
< # Clear the partition labels before running the zerombr and clearpart commands
< #  
< allparts=$( sfdisk -d | grep "/dev/"| grep -v "^#" | cut -d ":" -f1 | sed 's/[0-9]*//g' | sort | uniq )
< for part in $allparts; do
<     /sbin/parted -s -- $part mklabel gpt
< done
< 
```

After this executed, the sdj and sdk drives looks "clean" 
but now all the PowerKVM partitions appear back from the /dev/mapper ...  

```
[anaconda root@c910f05c39 tmp]# parted
GNU Parted 3.1
Using /dev/mapper/ibmpkvm_rootvg00-ibmpkvm_lv_data
Welcome to GNU Parted! Type 'help' to view a list of commands.
(parted) print all                                                        
Model: Linux device-mapper (linear) (dm)
Disk /dev/mapper/ibmpkvm_rootvg00-ibmpkvm_lv_data: 959GB
Sector size (logical/physical): 512B/4096B
Partition Table: loop
Disk Flags: 

Number  Start  End    Size   File system  Flags
 1      0.00B  959GB  959GB  ext4


Model: Linux device-mapper (linear) (dm)
Disk /dev/mapper/ibmpkvm_rootvg00-ibmpkvm_lv_var: 10.7GB
Sector size (logical/physical): 512B/4096B
Partition Table: loop
Disk Flags: 

Number  Start  End     Size    File system  Flags
 1      0.00B  10.7GB  10.7GB  ext4


Model: Linux device-mapper (linear) (dm)
Disk /dev/mapper/ibmpkvm_rootvg00-ibmpkvm_lv_swap: 8590MB
Sector size (logical/physical): 512B/4096B
Partition Table: loop
Disk Flags: 

Number  Start  End     Size    File system     Flags
 1      0.00B  8590MB  8590MB  linux-swap(v1)


Model: Linux device-mapper (linear) (dm)
Disk /dev/mapper/ibmpkvm_rootvg00-ibmpkvm_lv_system: 21.5GB
Sector size (logical/physical): 512B/4096B
Partition Table: loop
Disk Flags: 

Number  Start  End     Size    File system  Flags
 1      0.00B  21.5GB  21.5GB  ext4


Model: Linux device-mapper (linear) (dm)
Disk /dev/mapper/live-base: 2147MB
Sector size (logical/physical): 512B/512B
Partition Table: loop
Disk Flags: 

Number  Start  End     Size    File system  Flags
 1      0.00B  2147MB  2147MB  ext4


Model: Linux device-mapper (snapshot) (dm)
Disk /dev/mapper/live-rw: 2147MB
Sector size (logical/physical): 512B/512B
Partition Table: loop
Disk Flags: 

Number  Start  End     Size    File system  Flags
 1      0.00B  2147MB  2147MB  ext4


Model: ATA ST1000NX0313 (scsi)
Disk /dev/sdj: 1000GB
Sector size (logical/physical): 512B/4096B
Partition Table: gpt
Disk Flags: 

Number  Start  End  Size  File system  Name  Flags


Model: ATA ST1000NX0313 (scsi)
Disk /dev/sdk: 1000GB
Sector size (logical/physical): 512B/4096B
Partition Table: gpt
Disk Flags: 

Number  Start  End  Size  File system  Name  Flags


(parted)                                           
```

**Comment by whowutwut** [comment]
I also tried to brute force remove the devices in /dev/mapper...
```
#
# remove all the devices in /dev/mapper
#
dmsetup remove_all -f 
```

This cause the installer to crash, so this is not correct :)

**Comment by xuweibj** [comment]
@whowutwut 
The problem in step 3 is same with issue #821 ,  after work around rh7.2 installed successfully.

**Comment by xuweibj** [comment]
@whowutwut 

I tried several times to reproduce the problem (from step 1 to step 2), but did not occur.
This is log when install powerkvm3.1(some unimportant info of disk I deleted).
```
+ install_disk=
+ rm -f /tmp/install_disk
+ tmpdir=/tmp/xcat.getinstalldisk
+ mkdir -p /tmp/xcat.getinstalldisk
++ find /usr/bin /usr/etc /usr/games /usr/include /usr/lib /usr/lib64 /usr/libexec /usr/local /usr/sbin /usr/share /usr/src /usr/tmp -name awk
+ has_awk='/usr/bin/awk
/usr/libexec/awk
/usr/share/awk'
+ utolcmd='sed -e y/ABCDEFGHIJKLMNOPQRSTUVWXYZ/abcdefghijklmnopqrstuvwxyz/'
+ tmpfile=/tmp/xcat.getinstalldisk/getinstalldisk_
+ '[' -z '' ']'
+ echo '[get_install_disk]Information from /proc/partitions:'
[get_install_disk]Information from /proc/partitions:
+ cat /proc/partitions
major minor  #blocks  name

   7        0     433348 loop0
   7        1    2048000 loop1
   7        2     524288 loop2
   8        0  976762584 sda
   8       16  976762584 sdb
   8       17       7168 sdb1
   8       18  974753792 sdb2
   8       19    1999872 sdb3
  11        0    1048575 sr0
  11        1    1048575 sr1
  11        2    1048575 sr2
  11        3    1048575 sr3
 253        0    2048000 dm-0
 253        1    2048000 dm-1
+ echo ''

+ '[' -z '/usr/bin/awk
/usr/libexec/awk
/usr/share/awk' ']'
++ awk -F ' ' '{print $4}' /proc/partitions
++ grep -v name
++ grep -e '[s|h]d.*$'
+ entries='sda
sdb
sdb1
sdb2
sdb3'
+ for entry in '$entries'
+ '[' -z '/usr/bin/awk
/usr/libexec/awk
/usr/share/awk' ']'
++ udevadm info --query=property --name=/dev/sda
++ grep -i DEVTYPE
++ awk -F = '{print $2}'
++ sed -e y/ABCDEFGHIJKLMNOPQRSTUVWXYZ/abcdefghijklmnopqrstuvwxyz/
+ dev_type=disk
+ '[' disk == disk ']'
+ disks='sda '
+ for entry in '$entries'
+ '[' -z '/usr/bin/awk
/usr/libexec/awk
/usr/share/awk' ']'
++ grep -i DEVTYPE
++ udevadm info --query=property --name=/dev/sdb
++ awk -F = '{print $2}'
++ sed -e y/ABCDEFGHIJKLMNOPQRSTUVWXYZ/abcdefghijklmnopqrstuvwxyz/
+ dev_type=disk
+ '[' disk == disk ']'
+ disks='sda sdb '
+ for entry in '$entries'
+ '[' -z '/usr/bin/awk
/usr/libexec/awk
/usr/share/awk' ']'
++ udevadm info --query=property --name=/dev/sdb1
++ grep -i DEVTYPE
++ awk -F = '{print $2}'
++ sed -e y/ABCDEFGHIJKLMNOPQRSTUVWXYZ/abcdefghijklmnopqrstuvwxyz/
+ dev_type=partition
+ '[' partition == disk ']'
+ '[' partition == partition ']'
+ partitions='sdb1 '
+ for entry in '$entries'
+ '[' -z '/usr/bin/awk
/usr/libexec/awk
/usr/share/awk' ']'
++ udevadm info --query=property --name=/dev/sdb2
++ grep -i DEVTYPE
++ awk -F = '{print $2}'
++ sed -e y/ABCDEFGHIJKLMNOPQRSTUVWXYZ/abcdefghijklmnopqrstuvwxyz/
+ dev_type=partition
+ '[' partition == disk ']'
+ '[' partition == partition ']'
+ partitions='sdb1 sdb2 '
+ for entry in '$entries'
+ '[' -z '/usr/bin/awk
/usr/libexec/awk
/usr/share/awk' ']'
++ udevadm info --query=property --name=/dev/sdb3
++ grep -i DEVTYPE
++ awk -F = '{print $2}'
++ sed -e y/ABCDEFGHIJKLMNOPQRSTUVWXYZ/abcdefghijklmnopqrstuvwxyz/
+ dev_type=partition
+ '[' partition == disk ']'
+ '[' partition == partition ']'
+ partitions='sdb1 sdb2 sdb3 '
+ mount_dir=/tmp/xcat.getinstalldisk/xcat.getinstalldisk.mount
+ mkdir -p /tmp/xcat.getinstalldisk/xcat.getinstalldisk.mount
+ disk_array=
+ for partition in '$partitions'
+ echo '[get_install_disk]Check the partition sdb1.'
[get_install_disk]Check the partition sdb1.
+ '[' -e /tmp/xcat.getinstalldisk/getinstalldisk_sdb ']'
+ '[' -z '/usr/bin/awk
/usr/libexec/awk
/usr/share/awk' ']'
++ udevadm info --query=property --name=/dev/sdb1
++ grep -i FS_TYPE
++ awk -F = '{print $2}'
+ fs_type=
+ rc=255
+ '[' -z '' ']'
+ mount /dev/sdb1 /tmp/xcat.getinstalldisk/xcat.getinstalldisk.mount
mount: /dev/sdb1 is write-protected, mounting read-only
mount: wrong fs type, bad option, bad superblock on /dev/sdb1,
       missing codepage or helper program, or other error

       In some cases useful info is found in syslog - try
       dmesg | tail or so.
+ rc=32
+ '[' 32 -eq 0 ']'
+ echo '[get_install_disk]    Partition sdb1 mount failed or the partition is swap.'
[get_install_disk]    Partition sdb1 mount failed or the partition is swap.
+ for partition in '$partitions'
+ echo '[get_install_disk]Check the partition sdb2.'
[get_install_disk]Check the partition sdb2.
+ '[' -e /tmp/xcat.getinstalldisk/getinstalldisk_sdb ']'
+ '[' -z '/usr/bin/awk
/usr/libexec/awk
/usr/share/awk' ']'
++ udevadm info --query=property --name=/dev/sdb2
++ grep -i FS_TYPE
++ awk -F = '{print $2}'
+ fs_type=ext4
+ rc=255
+ '[' -z ext4 ']'
+ '[' ext4 '!=' swap ']'
+ mount -t ext4 /dev/sdb2 /tmp/xcat.getinstalldisk/xcat.getinstalldisk.mount
+ rc=0
+ '[' 0 -eq 0 ']'
+ echo '[get_install_disk]    Partition sdb2 mount success.'
[get_install_disk]    Partition sdb2 mount success.
+ ker_dir=/tmp/xcat.getinstalldisk/xcat.getinstalldisk.mount
+ '[' -d /tmp/xcat.getinstalldisk/xcat.getinstalldisk.mount/boot ']'
+ ker_dir=/tmp/xcat.getinstalldisk/xcat.getinstalldisk.mount/boot
+ for i in '$ker_dir/vmlinu*'
+ disk_part=sdb
+ touch /tmp/xcat.getinstalldisk/getinstalldisk_sdb
+ disk_array='sdb '
+ echo '[get_install_disk]    The partition sdb2 has kernel file.'
[get_install_disk]    The partition sdb2 has kernel file.
+ break
+ umount /tmp/xcat.getinstalldisk/xcat.getinstalldisk.mount
+ for partition in '$partitions'
+ echo '[get_install_disk]Check the partition sdb3.'
[get_install_disk]Check the partition sdb3.
+ '[' -e /tmp/xcat.getinstalldisk/getinstalldisk_sdb ']'
+ echo '[get_install_disk]    The disk sdb had OS installed, check next partition.'
[get_install_disk]    The disk sdb had OS installed, check next partition.
+ continue
+ '[' 'sdb ' ']'
+ disks='sdb '
+ echo '[get_install_disk]The disks which have kernel:'
[get_install_disk]The disks which have kernel:
+ echo '[get_install_disk]    sdb '
[get_install_disk]    sdb
+ echo ''

+ rmdir /tmp/xcat.getinstalldisk/xcat.getinstalldisk.mount
+ rm /tmp/xcat.getinstalldisk/getinstalldisk_sdb
+ has_wwn=0
+ has_path=0
+ file_pre=
+ disk_data=
+ for disk in '$disks'
++ udevadm info --query=property --name=sdb
+ disk_info='..............................disk info.....................................'
++ IFS=
++ echo '............................disk info.........................................'
++ grep '\<ID_WWN\>'
++ cut -d = -f2
+ output_for_wwn=0x5000c50080dfe825
++ echo 0x5000c50080dfe825
++ sed -e y/ABCDEFGHIJKLMNOPQRSTUVWXYZ/abcdefghijklmnopqrstuvwxyz/
+ disk_wwn=0x5000c50080dfe825
++ IFS=
++ grep DEVPATH
++ echo 'DEVLINKS...................disk info...........................................'
++ cut -d = -f2
+ output_for_path=/devices/pci0003:00/0003:00:00.0/0003:01:00.0/0003:02:0a.0/0003:0a:00.0/ata2/host1/target1:0:0/1:0:0:0/block/sdb
++ echo /devices/pci0003:00/0003:00:00.0/0003:01:00.0/0003:02:0a.0/0003:0a:00.0/ata2/host1/target1:0:0/1:0:0:0/block/sdb
++ sed -e y/ABCDEFGHIJKLMNOPQRSTUVWXYZ/abcdefghijklmnopqrstuvwxyz/
+ disk_path=/devices/pci0003:00/0003:00:00.0/0003:01:00.0/0003:02:0a.0/0003:0a:00.0/ata2/host1/target1:0:0/1:0:0:0/block/sdb
++ udevadm info --attribute-walk --name=sdb
++ grep -v '""'
++ grep -v '"sd"'
++ sed -e y/ABCDEFGHIJKLMNOPQRSTUVWXYZ/abcdefghijklmnopqrstuvwxyz/
++ head -n 1
++ sed -e 's/[^"]*"//' -e 's/"//'
++ grep DRIVERS
+ disk_driver=ahci
+ echo '[get_install_disk]The disk sdb information: '
[get_install_disk]The disk sdb information:
+ echo '[get_install_disk]    disk_wwn=0x5000c50080dfe825'
[get_install_disk]    disk_wwn=0x5000c50080dfe825
+ echo '[get_install_disk]    disk_path=/devices/pci0003:00/0003:00:00.0/0003:01:00.0/0003:02:0a.0/0003:0a:00.0/ata2/host1/target1:0:0/1:0:0:0
/block/sdb'
[get_install_disk]    disk_path=/devices/pci0003:00/0003:00:00.0/0003:01:00.0/0003:02:0a.0/0003:0a:00.0/ata2/host1/target1:0:0/1:0:0:0/block/s
db
+ echo '[get_install_disk]    disk_driver=ahci'
[get_install_disk]    disk_driver=ahci
+ '[' 0x5000c50080dfe825 ']'
+ has_wwn=1
+ file_pre=wwn
+ disk_data=0x5000c50080dfe825
+ case "$disk_driver" in
+ echo 'sdb 0x5000c50080dfe825'
+ echo '[get_install_disk]    Add disk: sdb 0x5000c50080dfe825 into wwn firstchoicedisks'
[get_install_disk]    Add disk: sdb 0x5000c50080dfe825 into wwn firstchoicedisks
+ for seq in first second third
+ '[' -s /tmp/xcat.getinstalldisk/getinstalldisk_wwnfirstchoicedisks ']'
+ install_file=/tmp/xcat.getinstalldisk/getinstalldisk_wwnfirstchoicedisks
+ break
+ '[' /tmp/xcat.getinstalldisk/getinstalldisk_wwnfirstchoicedisks ']'
+ '[' -s /tmp/xcat.getinstalldisk/getinstalldisk_wwnfirstchoicedisks ']'
++ cat /tmp/xcat.getinstalldisk/getinstalldisk_wwnfirstchoicedisks
++ grep -v '^$'
++ cut -d ' ' -f1
++ head -n 1
++ sort -k 2 -b
+ install_disk=/dev/sdb
+ echo '[get_install_disk]The install_disk is /dev/sdb by sorting wwn and DRIVER.'
[get_install_disk]The install_disk is /dev/sdb by sorting wwn and DRIVER.
+ rm /tmp/xcat.getinstalldisk/getinstalldisk_wwnfirstchoicedisks
+ rm -rf /tmp/xcat.getinstalldisk
+ '[' -z /dev/sdb ']'
+ echo /dev/sdb
+ '[' -e /tmp/install_disk ']'
```

From the log, it found that /dev/sdb had installed Ubuntu, and choose it. That's logic is correct.

I guess before install powerkvm, there are 2 OS installed on c39. If so, it will choose the one with smallest WWN(sdb or sdk). But it's just a guess.

**Comment by whowutwut** [comment]
@xuweibj  Maybe it was caused by a series of installs using older code that contained some issues that we fixed? So now we cannot reproduce with the latest code?

**Comment by whowutwut** [comment]
@tingtli  is it possible to have the autotest provision different operating systems on bare metal nodes that have multiple disks?

**Comment by tingtli** [comment]
The test cases does not care about how many disk the bare metal nodes have.  We can use current test cases and write a simple scripts to  run different os provision.  The os version is specified in default.conf configuration file.

**Comment by whowutwut** [comment]
@tingtli  yes, if we have a testcase that runs different OS provisioning in different combinations... it would cover it.  But this test needs to be done on bare metal nodes that have more than a single disk...  otherwise it won't catch it.  

Will close this issue for now because I don't know how to re-create the problem. If we see disk selection problem again, can open another issue.
